common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -53,6 +53,43 @@ def get_queryset(self):
         # to Admin users
         return CourseEntitlement.objects.all().select_related('user').select_related('enrollment_course_run')
 
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        username = serializer.data.get('user')
+        user = User.objects.get(username=username)
+        entitlement = CourseEntitlement.objects.get(uuid=serializer.data.get('uuid'), user=user.id, expired_at=None)
```

Comment:
My Django knowledge is lacking - do you have to pass the user id here or can you also pass the user object?

Reply:
Can pass the whole user object

Reply:
(see later in this file)